src/components/ui/input.tsx

```diff
@@ -24,20 +24,54 @@ const inputVariants = cva(
 
 export interface InputProps
   extends React.InputHTMLAttributes<HTMLInputElement>,
-    VariantProps<typeof inputVariants> {}
+    VariantProps<typeof inputVariants> {
+  startAdornment?: JSX.Element
+  endAdornment?: JSX.Element
+}
 export interface NumericalInputProps extends Omit<InputProps, 'onChange'> {
   onChange(value: string): void
 }
 
 const Input = React.forwardRef<HTMLInputElement, InputProps>(
-  ({ className, variant, type, ...props }, ref) => {
+  (
+    { className, variant, type, startAdornment, endAdornment, ...props },
+    ref
+  ) => {
+    const hasAdornment = Boolean(startAdornment) || Boolean(endAdornment)
     return (
-      <input
-        type={type}
-        className={cn(inputVariants({ variant, className }))}
-        ref={ref}
-        {...props}
-      />
+      <>
+        {hasAdornment ? (
```

Comment:
Move this if outside of the return, and the component outside of the Input

if (hasAdornment) ...
<AdornmentInput />
else 
<Input />